src/queue/SoundQueue.ts

```diff
@@ -0,0 +1,61 @@
+import config from '../../config/config.json';
+
+import QueueItem from './QueueItem';
+import Util from '../Util';
+
+export default class SoundQueue {
+  private queue: Array<QueueItem>;
+
+  constructor() {
+    this.queue = [];
+  }
+
+  public add(item: QueueItem) {
+    this.queue.push(item);
+  }
+
+  public isStartable() {
+    return this.queue.length === 1;
+  }
+
+  public start() {
+    this.playNext();
+  }
+
+  public clear() {
+    this.queue.length = 0;
+  }
+
+  public isEmpty() {
+    return this.queue.length === 0;
+  }
+
+  public current() {
+    return this.queue[0];
```

Comment:
I would argue that this should be moved to a field on the class instead of having at as the first element of the array.